### PR TITLE
BTS-1045: Bug fix/split modify test suite

### PR DESCRIPTION
**Pull request by dothebart**
### Scope & Purpose

Split aql-modify-noncluster testsuite into more digesteable parts

- [x] :hankey: Bugfix
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*

### CODE REVIEW

File: tests/js/server/aql/aql-modify-Insert-noncluster.js [hunk 1]
```diff
@@ -0,0 +1,808 @@
...
+      assertTrue(isEqual(collection.document(results[index]._key), results[index]));
+    }
+  }
+};
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief check whether the documents reported deleted are really gone
+////////////////////////////////////////////////////////////////////////////////
+
+var validateDeleteGone = function (collection, results) {
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Insert-noncluster.js:96]: seems to be unnecessary in this file

File: tests/js/server/aql/aql-modify-Insert-noncluster.js [hunk 2]
```diff
@@ -0,0 +1,808 @@
...
+      db._drop(cn2);
+      c1 = db._create(cn1);
+      c2 = db._create(cn2);
+
+      for (i = 0; i < 100; ++i) {
+        c1.save({ _key: "test" + i, value1: i, value2: "test" + i });
+      }
+      for (i = 0; i < 50; ++i) {
+        c2.save({ _key: "test" + i, value1: i, value2: "test" + i });
+      }
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Insert-noncluster.js:157]:
```suggestion
      db._drop(cn1);
      db._drop(cn2);
      c1 = db._create(cn1);
      c2 = db._create(cn2);

      let docs = [];
      for (let i = 0; i < 100; ++i) {
        docs.push({ _key: "test" + i, value1: i, value2: "test" + i });
      }
      c1.insert(docs);
      c2.insert(docs);
```

File: tests/js/server/aql/aql-modify-Insert-noncluster.js [hunk 3]
```diff
@@ -0,0 +1,808 @@
...
+    }
+  }
+};
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief convert flat document database to an associative array with the keys
+///        as object
+////////////////////////////////////////////////////////////////////////////////
+
+var wrapToKeys = function (results) {
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Insert-noncluster.js:117]: seems unnecessary in this file

File: tests/js/server/aql/aql-modify-Insert-noncluster.js [hunk 4]
```diff
@@ -0,0 +1,808 @@
...
+/// @brief check whether the documents inserted are equal on the db.
+////////////////////////////////////////////////////////////////////////////////
+
+var validateModifyResultInsert = function (collection, results) {
+  var index;
+  for (index in results) {
+    if (results.hasOwnProperty(index)){
+      assertTrue(isEqual(collection.document(results[index]._key), results[index]));
+    }
+  }
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Insert-noncluster.js:89]:
```suggestion
  for (let index in results) {
    if (results.hasOwnProperty(index)){
      assertTrue(isEqual(collection.document(results[index]._key), results[index]));
    }
  }
```

File: tests/js/server/aql/aql-modify-Insert-noncluster.js [hunk 5]
```diff
@@ -0,0 +1,808 @@
...
+  return stats;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief 
+////////////////////////////////////////////////////////////////////////////////
+
+var validateDocuments = function (documents, isEdgeCollection) {
+  var index;
+  for (index in documents) {
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Insert-noncluster.js:66]:
```suggestion
  for (let index in documents) {
```

File: tests/js/server/aql/aql-modify-Insert-noncluster.js [hunk 6]
```diff
@@ -0,0 +1,808 @@
...
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief executes the test suites
+////////////////////////////////////////////////////////////////////////////////
+
+jsunity.run(ahuacatlInsertSuite);
+
+return jsunity.done();
+
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Insert-noncluster.js:808]:
```suggestion

jsunity.run(ahuacatlInsertSuite);
return jsunity.done();
```

File: tests/js/server/aql/aql-modify-Modify-noncluster.js [hunk 7]
```diff
@@ -0,0 +1,879 @@
...
+  delete stats.peakMemoryUsage;
+  delete stats.intermediateCommits;
+  return stats;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief 
+////////////////////////////////////////////////////////////////////////////////
+
+var validateDocuments = function (documents, isEdgeCollection) {
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Modify-noncluster.js:64]: seems unnecessary here

File: tests/js/server/aql/aql-modify-Modify-noncluster.js [hunk 8]
```diff
@@ -0,0 +1,879 @@
...
+      }
+    }
+  }
+};
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief check whether the documents inserted are equal on the db.
+////////////////////////////////////////////////////////////////////////////////
+
+var validateModifyResultInsert = function (collection, results) {
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Modify-noncluster.js:83]: seems unnecessary here

File: tests/js/server/aql/aql-modify-Modify-noncluster.js [hunk 9]
```diff
@@ -0,0 +1,879 @@
...
+      assertTrue(isEqual(collection.document(results[index]._key), results[index]));
+    }
+  }
+};
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief check whether the documents reported deleted are really gone
+////////////////////////////////////////////////////////////////////////////////
+
+var validateDeleteGone = function (collection, results) {
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Modify-noncluster.js:96]: seems unnecessary here

File: tests/js/server/aql/aql-modify-Modify-noncluster.js [hunk 10]
```diff
@@ -0,0 +1,879 @@
...
+    }
+  }
+};
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief convert flat document database to an associative array with the keys
+///        as object
+////////////////////////////////////////////////////////////////////////////////
+
+var wrapToKeys = function (results) {
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Modify-noncluster.js:117]: seems unnecessary here

File: tests/js/server/aql/aql-modify-Modify-noncluster.js [hunk 11]
```diff
@@ -0,0 +1,879 @@
...
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief executes the test suites
+////////////////////////////////////////////////////////////////////////////////
+
+jsunity.run(ahuacatlModifySuite);
+
+return jsunity.done();
+
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Modify-noncluster.js:879]:
```suggestion
jsunity.run(ahuacatlModifySuite);
return jsunity.done();
```

File: tests/js/server/aql/aql-modify-Remove-noncluster.js [hunk 12]
```diff
@@ -0,0 +1,809 @@
...
+  return stats;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief 
+////////////////////////////////////////////////////////////////////////////////
+
+var validateDocuments = function (documents, isEdgeCollection) {
+  var index;
+  for (index in documents) {
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Remove-noncluster.js:66]:
```suggestion
  for (let index in documents) {
```

File: tests/js/server/aql/aql-modify-Remove-noncluster.js [hunk 13]
```diff
@@ -0,0 +1,809 @@
...
+      }
+    }
+  }
+};
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief check whether the documents inserted are equal on the db.
+////////////////////////////////////////////////////////////////////////////////
+
+var validateModifyResultInsert = function (collection, results) {
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Remove-noncluster.js:83]: unnecessary here

File: tests/js/server/aql/aql-modify-Remove-noncluster.js [hunk 14]
```diff
@@ -0,0 +1,809 @@
...
+  for (index in results) {
+    if (results.hasOwnProperty(index)){
+      try {
+        assertEqual(collection.document(results[index]._key), {});
+        fail();
+      }
+      catch (e) {
+        assertTrue(e.errorNum !== undefined, "unexpected error format while calling checking for deleted entry");
+        assertEqual(errors.ERROR_ARANGO_DOCUMENT_NOT_FOUND.code, e.errorNum, "unexpected error code (" + e.errorMessage + "): ");
+      }
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Remove-noncluster.js:107]:
```suggestion
  for (let index in results) {
    if (results.hasOwnProperty(index)) {
      try {
        assertEqual(collection.document(results[index]._key), {});
        fail();
      } catch (e) {
        assertNotUndefined(e.errorNum, "unexpected error format while calling checking for deleted entry");
        assertEqual(errors.ERROR_ARANGO_DOCUMENT_NOT_FOUND.code, e.errorNum, "unexpected error code (" + e.errorMessage + "): ");
      }
```

File: tests/js/server/aql/aql-modify-Remove-noncluster.js [hunk 15]
```diff
@@ -0,0 +1,809 @@
...
+    }
+  }
+};
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief convert flat document database to an associative array with the keys
+///        as object
+////////////////////////////////////////////////////////////////////////////////
+
+var wrapToKeys = function (results) {
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Remove-noncluster.js:117]: unnecessary here

File: tests/js/server/aql/aql-modify-Remove-noncluster.js [hunk 16]
```diff
@@ -0,0 +1,809 @@
...
+      db._drop(cn2);
+      c1 = db._create(cn1);
+      c2 = db._create(cn2);
+
+      for (i = 0; i < 100; ++i) {
+        c1.save({ _key: "test" + i, value1: i, value2: "test" + i });
+      }
+      for (i = 0; i < 50; ++i) {
+        c2.save({ _key: "test" + i, value1: i, value2: "test" + i });
+      }
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Remove-noncluster.js:157]:
```suggestion
      db._drop(cn1);
      db._drop(cn2);
      c1 = db._create(cn1);
      c2 = db._create(cn2);

      let docs = [];
      for (let i = 0; i < 100; ++i) {
        docs.push({ _key: "test" + i, value1: i, value2: "test" + i });
      }
      c1.insert(docs);
      c2.insert(docs.slice(0, 50));
```

File: tests/js/server/aql/aql-modify-Remove-noncluster.js [hunk 17]
```diff
@@ -0,0 +1,809 @@
...
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief executes the test suites
+////////////////////////////////////////////////////////////////////////////////
+
+jsunity.run(ahuacatlRemoveSuite);
+
+return jsunity.done();
+
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Remove-noncluster.js:809]:
```suggestion
jsunity.run(ahuacatlRemoveSuite);
return jsunity.done();
```

File: tests/js/server/aql/aql-modify-Update-noncluster.js [hunk 18]
```diff
@@ -0,0 +1,1984 @@
...
+  return stats;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief 
+////////////////////////////////////////////////////////////////////////////////
+
+var validateDocuments = function (documents, isEdgeCollection) {
+  var index;
+  for (index in documents) {
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Update-noncluster.js:66]:
```suggestion
  for (let index in documents) {
```

File: tests/js/server/aql/aql-modify-Update-noncluster.js [hunk 19]
```diff
@@ -0,0 +1,1984 @@
...
+};
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief check whether the documents inserted are equal on the db.
+////////////////////////////////////////////////////////////////////////////////
+
+var validateModifyResultInsert = function (collection, results) {
+  var index;
+  for (index in results) {
+    if (results.hasOwnProperty(index)){
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Update-noncluster.js:86]:
```suggestion
  for (let index in results) {
    if (results.hasOwnProperty(index)) {
```

File: tests/js/server/aql/aql-modify-Update-noncluster.js [hunk 20]
```diff
@@ -0,0 +1,1984 @@
...
+      assertTrue(isEqual(collection.document(results[index]._key), results[index]));
+    }
+  }
+};
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief check whether the documents reported deleted are really gone
+////////////////////////////////////////////////////////////////////////////////
+
+var validateDeleteGone = function (collection, results) {
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Update-noncluster.js:96]: unnecessary here

File: tests/js/server/aql/aql-modify-Update-noncluster.js [hunk 21]
```diff
@@ -0,0 +1,1984 @@
...
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief convert flat document database to an associative array with the keys
+///        as object
+////////////////////////////////////////////////////////////////////////////////
+
+var wrapToKeys = function (results) {
+  var keyArray = {};
+  var index;
+  for (index in results) {
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Update-noncluster.js:120]:
```suggestion
  for (let index in results) {
```

File: tests/js/server/aql/aql-modify-Update-noncluster.js [hunk 22]
```diff
@@ -0,0 +1,1984 @@
...
+      c2 = db._create(cn2);
+      c3 = db._create(cn3);
+
+      for (i = 0; i < 100; ++i) {
+        c1.save({ _key: "test" + i, value1: i, value2: "test" + i });
+      }
+      for (i = 0; i < 50; ++i) {
+        c2.save({ _key: "test" + i, value1: i, value2: "test" + i });
+      }
+      let docs = [];
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Update-noncluster.js:162]:
```suggestion
      db._drop(cn1);
      db._drop(cn2);
      db._drop(cn3);
      c1 = db._create(cn1);
      c2 = db._create(cn2);
      c3 = db._create(cn3);

      let docs = [];
      for (let i = 0; i < 100; ++i) {
        docs.push({ _key: "test" + i, value1: i, value2: "test" + i });
      }
      c1.insert(docs);
      c2.insert(docs.slice(0, 50));
      
      docs = [];
```

File: tests/js/server/aql/aql-modify-Update-noncluster.js [hunk 23]
```diff
@@ -0,0 +1,1984 @@
...
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief executes the test suites
+////////////////////////////////////////////////////////////////////////////////
+
+jsunity.run(ahuacatlUpdateSuite);
+
+return jsunity.done();
+
```
Comment (jsteemann) [inline tests/js/server/aql/aql-modify-Update-noncluster.js:1984]:
```suggestion
jsunity.run(ahuacatlUpdateSuite);
return jsunity.done();
```

**Comment by dothebart** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/24488/ green as well.

**Pull request merged**